opm/simulators/linalg/bda/BILU0.cpp

```diff
@@ -82,11 +91,26 @@ namespace bda
         } else if (opencl_ilu_reorder == ILUReorder::GRAPH_COLORING) {
             out << "BILU0 reordering strategy: " << "graph_coloring\n";
             findGraphColoring<block_size>(mat->colIndices, mat->rowPointers, CSCRowIndices, CSCColPointers, mat->Nb, mat->Nb, mat->Nb, &numColors, toOrder, fromOrder, rowsPerColor);
+        } else if (opencl_ilu_reorder == ILUReorder::NONE) {
+            out << "BILU0 reordering strategy: none\n";
+            numColors = 1;
+            rowsPerColor.emplace_back(Nb);
+            // numColors = Nb;
+            // for(int i = 0; i < Nb; ++i){
+            //     rowsPerColor.emplace_back(1);
+            // }
+            for(int i = 0; i < Nb; ++i){
+                toOrder[i] = i;
+                fromOrder[i] = i;
+            }
```

Comment:
The mapping arrays to- and fromOrder where needed, because the matrix `mat` was always reordered into `rmat`. Even when `NONE` was chosen. openclSolver uses `rmat` for the spmv, and with the right values in the mapping arrays, `rmat == mat`.

There is another issue here: `numColors==1` means that all rows are in the same color, and are applied (during the solve) in parallel. This loses accuracy when using the exact decomposition, but should be fine for the FGPILU algorithm.
`numColors==Nb` means that every row has their own color, so they are applied sequentially. This is extremely slow in runtime, but the convergence (number of lin iterations) seems ok.

Reply:
Addressed in 71d62950